### PR TITLE
Embedded broker support different partition counts

### PR DESCRIPTION
**Pull request by garyrussell**
Add `addTopics(NewTopic... topics)` to the embedded broker to enable
creation of topics with different parition counts.

### CODE REVIEW

File: src/reference/asciidoc/testing.adoc [hunk 1]
```diff
@@ -103,41 +103,65 @@ Convenient constants `EmbeddedKafkaBroker.SPRING_EMBEDDED_KAFKA_BROKERS` and `Em
...
+[source, java]
+----
+public class MyTests {
+
+    private static EmbeddedKafkaRule embeddedKafka = new EmbeddedKafkaRule(1, false, 5, "foo", "bar");
+
+    @Test
+    public void test() {
+        embeddedKafkaRule.getEmbeddedKafka()
+              .addTopics(new NewTopic("baz", 10, (short) 1), new NewTopic("qux", 5, (short) 1));
```
Comment (artembilan) [inline src/reference/asciidoc/testing.adoc]: Please, fix this for `15` as you explain below.

Comment (artembilan) [reply src/reference/asciidoc/testing.adoc]: OK. I guess you are busy with something else.
So, fixing on merge...

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java [hunk 2]
```diff
@@ -87,8 +88,11 @@
 	int partitions() default 2;
 
 	/**
-	 * Topics that should be created
-	 * Topics may contain property placeholders, e.g. {@code topics = "${kafka.topic.one:topicOne}"}
+	 * Topics that should be created Topics may contain property placeholders, e.g.
+	 * {@code topics = "${kafka.topic.one:topicOne}"} The topics will be created with
+	 * {@link #partitions()} partitions; to provision other topics with other partition
+	 * counts call the {@code addPartitions(NewTopic... topics)} method on the autowired
```
Comment (artembilan) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java]: Must be `addTopics()`.
Fixing one merge...

Comment (garyrussell) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java]: thx

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/adc94630582dc037cce5b47745093d247632d605

**Pull request closed without merging**